packages/opentelemetry-sdk-trace-base/src/export/SimpleSpanProcessor.ts

```diff
@@ -35,18 +35,30 @@ import { SpanExporter } from './SpanExporter';
  */
 export class SimpleSpanProcessor implements SpanProcessor {
   private _shutdownOnce: BindOnceFuture<void>;
+  private _unresolvedResources: Set<Promise<void>>;
 
   constructor(private readonly _exporter: SpanExporter) {
     this._shutdownOnce = new BindOnceFuture(this._shutdown, this);
+    this._unresolvedResources = new Set<Promise<void>>();
   }
 
-  forceFlush(): Promise<void> {
-    // do nothing as all spans are being exported without waiting
-    return Promise.resolve();
+  async forceFlush(): Promise<void> {
+    // await unresolved resources before resolving
+    await Promise.all(Array.from(this._unresolvedResources));
   }
 
-  // does nothing.
-  onStart(_span: Span, _parentContext: Context): void {}
+  onStart(_span: Span, _parentContext: Context): void {
+    // store the resource's unresolved promise
+    if (!_span.resource.asyncAttributesHaveResolved) {
+      const resourcePromise = _span.resource.waitForAsyncAttributes();
```

Comment:
I suspect we might run into similar issues of interoperability if a new processor receives resources from a previous implementation.  This is a valid setup users might craft and it should work according to semver guarantees

In this case, the resource object will not implement the `waitForAsyncAttributes` function and this code might throw.

I guess to make it compatible, the code needs to be prepared so that `asyncAttributesHaveResolved` and `waitForAsyncAttributes` are not found on the resource object